tools/python/m2k-osc/ain_tab.py

```diff
@@ -0,0 +1,169 @@
+        ain_range_label.grid(row=12,column=0)
+        hystrange_label=tk.Label(self,text="Hysteresis Range",justify="left",anchor='w',width=20,font=SMALL_FONT)
+        hystrange_label.grid(row=10,column=0)
+        sclfact_label=tk.Label(self,text="Scaling Factor",justify="left",anchor='w',width=20,font=SMALL_FONT)
+        sclfact_label.grid(row=11,column=0)
+        hystrangech0_box=tk.Label(self,textvariable=self.hystrange_ch0)
+        hystrangech0_box.grid(row=10,column=1)
+        hystrangech1_box=tk.Label(self,textvariable=self.hystrange_ch1)
+        hystrangech1_box.grid(row=10,column=2)
+        sclfactch0_box=tk.Label(self,textvariable=self.sclfact_ch0)
+        sclfactch0_box.grid(row=11,column=1)
+        sclfactch1_box=tk.Label(self,textvariable=self.sclfact_ch1)
+        sclfactch1_box.grid(row=11,column=2)
+        self.ain_range_ch0_var= tk.IntVar()
+        ain_range_ch0_check=tk.Checkbutton(self,text ="Set High Range",variable=self.ain_range_ch0_var,command=self.set_range_ch0,font=SMALL_FONT)
+        ain_range_ch0_check.grid(row=12,column=1)
+        self.ain_range_ch1_var= tk.IntVar()
+        ain_range_ch1_check=tk.Checkbutton(self,text ="Set High Range",variable=self.ain_range_ch1_var,command=self.set_range_ch1,font=SMALL_FONT)
+        ain_range_ch1_check.grid(row=12,column=2)
+        ain_rangelimits_label=tk.Label(self,text="Range Limits",justify="left",anchor='w',width=20,font=SMALL_FONT)
+        ain_rangelimits_label.grid(row=13,column=0)
+        self.rangelimits_ch0=tk.StringVar()
+        self.rangelimits_ch1=tk.StringVar()
+        self.rangelimits_ch0.set(str(self.ain.getRangeLimits(self.ain.getRange(0))))
+        self.rangelimits_ch1.set(str(self.ain.getRangeLimits(self.ain.getRange(1))))
+        ain_rangelimits_ch0=tk.Label(self,textvariable=self.rangelimits_ch0,font=SMALL_FONT)
+        ain_rangelimits_ch0.grid(row=13,column=1)
+        ain_rangelimits_ch1=tk.Label(self,textvariable=self.rangelimits_ch1,font=SMALL_FONT)
+        ain_rangelimits_ch1.grid(row=13,column=2)
+
+    def change_sample_rate(self):
+        ain=self.ain
+        desired_samplerate=float(self.in_samplerate.get())
+        ain.setSampleRate(desired_samplerate)
+        self.filter_comp.set(str(ain.getFilterCompensation(ain.getSampleRate())))
+
+    def get_values(self):
+        ain=self.ain
+        self.calibscale_ch0.set(str(ain.getCalibscale(0)))
+        self.calibscale_ch1.set(str(ain.getCalibscale(1)))
+        self.vertoffset_ch0.set(str(ain.getVerticalOffset(0)))
+        self.vertoffset_ch1.set(str(ain.getVerticalOffset(1)))
+        self.rangelimits_ch0.set(str(ain.getRangeLimits(ain.getRange(0))))
+        self.rangelimits_ch1.set(str(ain.getRangeLimits(ain.getRange(1))))
+        self.hystrange_ch0.set(str(ain.getHysteresisRange(0)))
+        self.hystrange_ch1.set(str(ain.getHysteresisRange(1)))
+        self.sclfact_ch0.set(str(ain.getScalingFactor(0)))
+        self.sclfact_ch1.set(str(ain.getScalingFactor(1)))
```

Comment:
i'm wondering if it would be a good idea to reverse the logic on this ch0 & ch1 attributes;
maybe group them somehow [a class or array, or hashtable]
then you could have some way like  `self.ch0.get_values()` & `self.ch1.get_values()`
you could also construct the ch0 & ch1 members with a single constructor;
`self.ch0 = self.init_ch(0)` or something like that;

i'm not saying this should be the exact code,but something along these lines would look more compact;